packages/slate-editor/src/extensions/button-block/ButtonBlockExtension.tsx

```diff
@@ -0,0 +1,45 @@
+import { createDeserializeElement, type Extension } from '@prezly/slate-commons';
+import React from 'react';
+import type { RenderElementProps } from 'slate-react';
+
+import { composeElementDeserializer } from '#modules/html-deserialization';
+
+import { ButtonBlockNode } from './ButtonBlockNode';
+import { ButtonBlockElement } from './components';
+import { normalizeRedundantButtonBlockAttributes, parseSerializedButtonBlockElement } from './lib';
+
+export const EXTENSION_ID = 'ButtonBlockExtension';
+
+export interface ButtonBlockExtensionConfiguration {
+    withNewTabOption?: boolean;
+}
+
+export function ButtonBlockExtension({
+    withNewTabOption = true,
+}: ButtonBlockExtensionConfiguration): Extension {
+    return {
+        id: EXTENSION_ID,
+        deserialize: {
+            element: composeElementDeserializer({
+                [ButtonBlockNode.Type]: createDeserializeElement(parseSerializedButtonBlockElement),
+            }),
+        },
+        normalizeNode: normalizeRedundantButtonBlockAttributes,
```

Comment:
I've realized that because we have `uuid` in the button node, we also want to make sure that it's unique within the document.

Let's add a normalization function for it. Please check `fixDuplicatePlaceholderUuid()` as an example.

https://github.com/prezly/slate/blob/d58e14a5ed5edf3ef9fb373345f038ae5ace3ad2/packages/slate-editor/src/extensions/placeholders/normalization/fixDuplicatePlaceholderUuid.ts#L7-L24

Reply:
updated